### DISCUSSION TITLE
Option for Quarto’s HTTP server to enforce case sensitivity of paths

### DISCUSSION THREAD

**Discussion by rben01** [quarto-dev/quarto-cli#8982] [Feature Requests]
### Description

`quarto preview` is a handy command to view your pages as they will be rendered when published (e.g., to GitLab Pages). However, unlike Quarto’s HTTP server, GitLab Pages (and I imagine many other sites) are case sensitive when serving URLs. This can lead to unexpected differences between a Quarto-previewed page and the published page, as files such as images and videos that do revolve when previewed locally may fail to resolve when viewed on the published site, if their case in the Quarto source does not match their case on the filesystem. For example, if you have the file `./img.JPG`, but write `![](img.jpg)`, then `quarto preview` will successfully serve the file, whereas GitLab Pages will not.

It would be nice if Quarto had an option to enforce case sensitivity of path/URL resolution so that these errors could be caught during the `preview` stage.

**Comment by cscheid** [comment]
This is honestly low in our priority list. Dealing with it correctly involves behavior that is dependent across deployments, operating systems and file systems (windows and macos are case insensitive-but-preserving by default. Linux is case sensitive. However, there can exist macos filesystems that _are_ case sensitive. Now multiply this by web server behavior)

If anything, I think the fix would be to warn or disallow any uppercase letters in filenames that are deployed; it's almost always a bad idea.
